### PR TITLE
[react-infinite] Upgrade react-infinite to 0.13.0

### PR DESCRIPTION
**Pull request by EmergentBehavior**
* Bump to React 16 (supports it now)
* Latest deps versions
* Validate checksums
* Proper cljsjs namespacing

### CODE REVIEW

File: react-infinite/build.boot [hunk 1]
```diff
@@ -21,17 +21,17 @@
...
+   (sift :move {#"^react-infinite-(.*)/dist/react-infinite.js$"
+                "cljsjs/react-infinite/development/react-infinite.inc.js"
+                #"^react-infinite-(.*)/dist/react-infinite.min.js$"
+                "cljsjs/react-infinite/production/react-infinite.min.inc.js"})
    (sift :include #{#"^cljsjs"})
-
-   (deps-cljs :name "react-infinite"
-              :requires ["react" "react-dom" "prop-types"])
+   (deps-cljs :name "cljsjs.react-infinite"
+              :requires ["cljsjs.react"
```
Comment (Deraen) [inline react-infinite/build.boot]: The names without `cljsjs.` prefix are in fact correct. These names are used to match npm packages, so the libraries can use either cljsjs foreign-lib or npm modules, without changing their code (if the foreign-lib also supports global-exports). For backwards compatibility e.g. React provides both `react` and `cljsjs.react`. Packages using global-exports should depend the "new" name (`react`), and others can choose either.

As this package used to provide `react-infinite` I think it is best to provide the same name in new version, but we can also provide the "old" style name, this should do:

```
(deps-cljs :foreign-libs [{:file #"react-infinite.inc.js"
                           :file-min #"react-infinite.min.inc.js"
                           :provides ["react-infinite" "cljsjs.react-infinite"
                           :requires ["react" "react-dom" "prop-types"]
    :externs [#"react-infinite.ext.js"])
```

File: react-infinite/build.boot [hunk 2]
```diff
@@ -21,17 +21,17 @@
...
+                         "cljsjs.react.dom"
+                         "cljsjs.prop-types"])
    (pom)
-
-   (jar)))
+   (jar)
+   (validate-checksums)))
```
Comment (Deraen) [inline react-infinite/build.boot]: Try using `validate` here for foreign lib and JS file validation, in addtion to checksums.

File: react-infinite/build.boot (hunk 1 above)
Comment (Deraen) [reply react-infinite/build.boot]: https://github.com/cljsjs/packages/wiki/Unified-module-names-and-global-exports

**Comment by EmergentBehavior** [comment]
@Deraen Thanks for taking a look! Will make both changes and push shortly.

If you're wondering, I'm upgrading many dependencies for one of my company's applications. That's why I'm firing off so many PRs!

**Comment by EmergentBehavior** [comment]
@Deraen should be good to go

**Pull request merged**